ossfuzz/fuzz_data_producer.c

```diff
@@ -32,24 +32,36 @@ uint32_t FUZZ_dataProducer_uint32(FUZZ_dataProducer_t *producer, uint32_t min,
     rolling >>= 8;
   }
 
-  if (range == 0xffffffff) {
-    return result;
-  }
+  return result;
+}
 
-  return min + result % (range + 1);
+uint32_t FUZZ_dataProducer_uint32(uint32_t seed, uint32_t min, uint32_t max)
+{
+    uint32_t range = max - min;
+    if (range == 0xffffffff) {
+      return seed;
+    }
+    return min + seed % (range + 1);
+}
+
+uint32_t FUZZ_dataProducer_uint32NonAdaptive(FUZZ_dataProducer_t* producer,
```

Comment:
Same idea, try to preserve API consistency.
What about `FUZZ_dataProducer_range32()` ?